parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetWriter.java

```diff
@@ -208,11 +225,7 @@ public Optional<ParquetValueWriters.PrimitiveWriter<?>> visit(
           LogicalTypeAnnotation.TimeUnit.MICROS.equals(timestampType.getUnit()),
           "Cannot write timestamp in %s, only MICROS is supported",
           timestampType.getUnit());
-      if (timestampType.isAdjustedToUTC()) {
-        return Optional.of(new TimestamptzWriter(desc));
-      } else {
-        return Optional.of(new TimestampWriter(desc));
-      }
+      return timestampWriter(desc, timestampType.isAdjustedToUTC());
```

Comment:
Should I pass a `TimeUnit` to avoid modifying the signature in the future when other units are supported?  Also same for timeType